components/Common/Search/States/WithSearchBox.tsx

```diff
@@ -78,16 +89,57 @@ export const WithSearchBox: FC<SearchBoxProps> = ({ onClose }) => {
     [searchTerm, selectedFacet]
   );
 
-  const reset = () => {
-    setSearchTerm('');
-    setSearchResults(null);
-    setSelectedFacet(0);
+  useKeyboardCommands(cmd => {
+    if (searchError || !searchResults || searchResults.count <= 0) {
+      return;
+    }
+
+    switch (true) {
+      case cmd === 'down' && selectedResult === undefined:
+        setSelectedResult(0);
+        break;
+      case cmd === 'down' &&
+        selectedResult != undefined &&
+        selectedResult < searchResults.count &&
+        selectedResult < DEFAULT_ORAMA_QUERY_PARAMS.limit - 1:
+        setSelectedResult(selectedResult + 1);
+        break;
+      case cmd === 'up' && selectedResult != undefined && selectedResult != 0:
+        setSelectedResult(selectedResult - 1);
+        break;
+      case cmd === 'enter':
+        handleEnter();
+        break;
+      default:
+    }
+  });
+
+  const handleEnter = () => {
+    if (
+      selectedResult === undefined ||
+      (searchResults && searchResults?.count <= 0)
+    ) {
+      return;
+    }
+    const selectedHit = searchResults?.hits[selectedResult ?? 0];
```

Comment:
```suggestion
    const selectedHit = searchResults.hits[selectedResult];
```

You are already checking for undefined on the other if statement

Reply:
True, changed